service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -92,15 +97,32 @@ private boolean validateRequestPath(DefaultHttpSession acceptSession) {
         return requestPath.startsWith(acceptPath);
     }
 
+
+    /**
+     * Helper method performing loop detection
+     * @param acceptSession - session parameter
+     */
+    private void performLoopDetection(DefaultHttpSession acceptSession) {
```

Comment:
May be it needs to return boolean so that no connect is performed when loop is detected.